### ISSUE TITLE
Detect unknown cfg predicates E0537

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#7096]
`cfg` attribute may have only `not`, `all` or `any` predicates according to [reference](https://doc.rust-lang.org/reference/conditional-compilation.html). Other ones are treated as [E0537](https://doc.rust-lang.org/error-index.html#E0537) errors by the compiler.

Currently, the plugin doesn't detect such errors, so you can easily write `non` instead of `not` and get quite weird behavior of the plugin. It would be great to highlight wrong predicates.

Part of #886

```rust
#[cfg(target_os = "android")]
fn foo() {}

#[cfg(non(target_os = "android"))] // <- error[E0537]: invalid predicate `non`
fn foo() {}

fn main() {
    foo();
}
```
